integrade/api.py

```diff
@@ -0,0 +1,216 @@
+# coding=utf-8
+"""Client for working with Cloudigrade's API.
+
+This module provides a flexible API client for talking with the cloudigrade
+server, allowing the user to customize how return codes are handled depending
+on the context.
+
+"""
+from json import JSONDecodeError
+from pprint import pformat
+from urllib.parse import urljoin, urlunparse
+
+from integrade import config, exceptions
+
+import requests
+from requests.exceptions import HTTPError
+
+
+def raise_error_for_status(response):
+    """Generate an error message and raise HTTPError for bad return codes.
+
+    :raises: ``requests.exceptions.HTTPError`` if the response status code is
+        in the 4XX or 5XX range.
+    """
+    r = response
```

Comment:
Why are we just renaming the input variable like this? Doing this obfuscates its meaning and is counter to the PEP8 guide:

https://www.python.org/dev/peps/pep-0008/#names-to-avoid

Reply:
Probably just trying to work around that 80 char limit will revisit